azurerm/resource_arm_virtual_machine.go

```diff
@@ -71,6 +71,25 @@ func resourceArmVirtualMachine() *schema.Resource {
 				},
 			},
 
+			"identity": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
```

Comment:
it'd be good to add some validation (and a case-insensitive check) for this field, even if there's only one allowable value at this time:

```
ValidateFunc: validation.StringInSlice([]string { "SystemAssigned", true),
DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
```